### DISCUSSION TITLE
Problem with calcGrains in MTex

### DISCUSSION THREAD

**Discussion by Sogo5030** [mtex-toolbox/mtex#2155] [Ask Anything]
Can someone give me some guide, if I use this line:

[grains,ebsd.grainId, ebsd.mis2mean] = calcGrains(ebsd,'angle',1*degree);

I get this error,

Undefined function 'sparse' for input arguments of type 'int32'.

Error in spatialDecomposition (line 44)
      I_FD = sparse(I_ED1(I_ED2<=numX),I_ED2(I_ED2<=numX),1,size(F,1),numX);

Error in EBSD/calcGrains (line 70)
  [V,F,I_FD] = spatialDecomposition([ebsd.prop.x(:), ebsd.prop.y(:)],ebsd.unitCell,varargin{:});

**Comment by kilir** [comment]
Hi,

does calcGrains work in other cases?
Can you elaborate a bit more on your Matlab version etc?

Cheers,
Rüdiger
